packages/sewing-kit-koa/src/tests/assets.test.ts

```diff
@@ -125,27 +169,122 @@ describe('Assets', () => {
+                scripts: [createMockAsset(scriptOne)],
+              }),
+            }),
+          }),
+          createMockManifestEntry({
+            manifest: createMockManifest({
+              main: createMockEntrypoint({
+                scripts: [createMockAsset(scriptTwo)],
+              }),
+            }),
+          }),
+        ]),
+      );
+
+      const assets = new Assets(defaultOptions);
+
+      expect(await assets.scripts()).toEqual([{path: scriptTwo}]);
+    });
+
+    it('uses the last manifest when no manifest matches', async () => {
+      readJson.mockImplementation(() =>
+        createMockConsolidatedManifest([
+          createMockManifestEntry({
+            browsers: ['firefox > 1'],
+            manifest: createMockManifest({
+              main: createMockEntrypoint({
+                scripts: [createMockAsset(scriptOne)],
+              }),
+            }),
+          }),
+          createMockManifestEntry({
+            browsers: ['safari > 1'],
+            manifest: createMockManifest({
+              main: createMockEntrypoint({
+                scripts: [createMockAsset(scriptTwo)],
+              }),
+            }),
+          }),
+        ]),
+      );
+
+      const assets = new Assets({...defaultOptions, userAgent: chrome71});
+
+      expect(await assets.scripts()).toEqual([{path: scriptTwo}]);
+    });
+
+    it('uses the first matching manifest', async () => {
+      readJson.mockImplementation(() =>
+        createMockConsolidatedManifest([
+          createMockManifestEntry({
+            browsers: ['chrome > 60'],
+            manifest: createMockManifest({
+              main: createMockEntrypoint({
+                scripts: [createMockAsset(scriptOne)],
+              }),
+            }),
+          }),
+          createMockManifestEntry({
+            browsers: ['chrome > 1'],
+            manifest: createMockManifest({
+              main: createMockEntrypoint({
+                scripts: [createMockAsset(scriptTwo)],
+              }),
+            }),
+          }),
+        ]),
+      );
+
+      const assets = new Assets({...defaultOptions, userAgent: chrome71});
+
+      expect(await assets.scripts()).toEqual([{path: scriptOne}]);
+    });
+  });
 });
 
-function createMockAssetList({
-  name = 'main',
-  styles = [],
+function createMockAsset(path: string): Asset {
+  return {path};
+}
+
+function createMockEntrypoint({
   scripts = [],
+  styles = [],
 }: {
-  name?: string;
-  styles?: string[];
-  scripts?: string[];
+  scripts?: Asset[];
+  styles?: Asset[];
 } = {}) {
-  return {
-    entrypoints: {
-      [name]: {
-        js: scripts.map(path => ({
-          path,
-        })),
-        css: styles.map(path => ({
-          path,
-        })),
-      },
-    },
-  };
+  return {js: scripts, css: styles};
+}
+
+function createMockManifest(entrypoints: {[key: string]: Entrypoint} = {}) {
+  return {entrypoints};
+}
+
+function createMockManifestEntry({
```

Comment:
We might want to expose these for external test usage / put them in their own package later.